truelearn/preprocessing/wikifier.py

```diff
@@ -0,0 +1,172 @@
+import orjson
+from urllib import request, parse, error
+
+from typing import Optional, Union, NoReturn
+Annotation = dict[str, Union[str, float, None]]
+WikifierResponse = dict[str, Union[list[str], list[dict]]]
+
+
+class Wikifier:
+    """
+    A class for making requests to the Wikifier API
+
+    Attributes
+    ----------
+    api_key: str
+        the API key needed to make the request, get one from
+        https://wikifier.org/register.html
+
+    Methods
+    -------
+    wikify(text, df_ignore, words_ignore)
+        Annotates input text using the Wikifier API.
+
+    References
+    ----------
+    .. [1] Janez Brank, Gregor Leban, Marko Grobelnik. Annotating Documents
+    with Relevant Wikipedia Concepts. Proceedings of the Slovenian Conference
+    on Data Mining and Data Warehouses (SiKDD 2017), Ljubljana, Slovenia,
+    9 October 2017.
+
+    """
+
+    def __init__(self, api_key: str) -> None:
+        if isinstance(api_key, str):
+            self.api_key = api_key
+        else:
+            raise TypeError("api_key should be a string")   
+
+    def wikify(
+            self, text: str, df_ignore: int = 50,
+            words_ignore: int = 50, top_n: Optional[int] = None
+        ) -> Union[list[Annotation], NoReturn]:

```

Comment:
I suspect that the use of NoReturn is incorrect. [NoReturn vs. None in "void" functions - type annotations in Python 3.6](https://stackoverflow.com/questions/48038149/noreturn-vs-none-in-void-functions-type-annotations-in-python-3-6)